From akpm@linux-foundation.org Tue Oct  2 14:41:03 2007
From: Andrew Morton <akpm@linux-foundation.org>
Date: Tue, 02 Oct 2007 14:40:46 -0700
Subject: USB: fix gregkh-usb-usb-sisusb2vga-convert-printk-to-dev_-macros
To: greg@kroah.com
Cc: linux-usb-devel@lists.sourceforge.net, akpm@linux-foundation.org, felipe.lima@indt.org.br, thomas@winischhofer.net
Message-ID: <200710022140.l92LekM7023764@imap1.linux-foundation.org>


From: Andrew Morton <akpm@linux-foundation.org>

drivers/usb/misc/sisusbvga/sisusb.c: In function 'sisusb_open':
drivers/usb/misc/sisusbvga/sisusb.c:2444: warning: 'sisusb' is used uninitialized in this function

I can tell that'll oops just by looking at it.

How come this code assume a 7,000 column xterm? :(

Cc: Felipe Balbi <felipe.lima@indt.org.br>
Cc: Thomas <thomas@winischhofer.net>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/misc/sisusbvga/sisusb.c |   10 +---------
 1 file changed, 1 insertion(+), 9 deletions(-)

--- a/drivers/usb/misc/sisusbvga/sisusb.c
+++ b/drivers/usb/misc/sisusbvga/sisusb.c
@@ -3323,15 +3323,7 @@ static int __init usb_sisusb_init(void)
 	sisusb_init_concode();
 #endif
 
-	if (!(retval = usb_register(&sisusb_driver))) {
-
-		dev_info(&sisusb->sisusb_dev->dev, "Driver version %d.%d.%d\n", SISUSB_VERSION,
-				SISUSB_REVISION, SISUSB_PATCHLEVEL);
-		dev_info(&sisusb->sisusb_dev->dev, "sisusb: Copyright (C) 2005 Thomas Winischhofer\n");
-
-	}
-
-	return retval;
+	return usb_register(&sisusb_driver);
 }
 
 static void __exit usb_sisusb_exit(void)
